docs/strategy-advanced.md

```diff
@@ -229,3 +229,83 @@ for val in self.buy_ema_short.range:
 # Append columns to existing dataframe
 merged_frame = pd.concat(frames, axis=1)
 ```
+
+## Adjust trade position
+
+`adjust_trade_position()` can be used to perform additional orders to manage risk with DCA (Dollar Cost Averaging) for example.
+The strategy is expected to return a stake_amount if and when an additional buy order should be made (position is increased).
+If there is not enough funds in the wallet then nothing will happen.
+Additional orders also mean additional fees and those orders don't count towards `max_open_trades`.
+Unlimited stake amount with trade position increasing is highly not recommended as your DCA orders would compete with your normal trade open orders.
+
+!!! Note
+    The `position_adjustment_enable` configuration parameter must be enabled to use adjust_trade_position callback in strategy.
+
+!!! Warning
+    Stoploss is still calculated from the initial opening price, not averaged price.
+
+``` python
+from freqtrade.persistence import Trade
+
+
+class DigDeeperStrategy(IStrategy):
+    
+    # Attempts to handle large drops with DCA. High stoploss is required.
+    stoploss = -0.30
+    
+    # ... populate_* methods
+
+    def adjust_trade_position(self, pair: str, trade: Trade,
+                              current_time: datetime, current_rate: float, current_profit: float,
+                                  **kwargs) -> Optional[float]:
+        """
+        Custom trade adjustment logic, returning the stake amount that a trade should be increased.
+        This means extra buy orders with additional fees.
+ 
+        :param pair: Pair that's currently analyzed
+        :param trade: trade object.
+        :param current_time: datetime object, containing the current datetime
+        :param current_rate: Rate, calculated based on pricing settings in ask_strategy.
+        :param current_profit: Current profit (as ratio), calculated based on current_rate.
+        :param **kwargs: Ensure to keep this here so updates to this won't break your strategy.
+        :return float: Stake amount to adjust your trade
+        """
+
+        if current_profit > -0.05:
+            return None
+
+        # Obtain pair dataframe.
+        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
+            
+        # Only buy when not actively falling price.
+        last_candle = dataframe.iloc[-1].squeeze()
+        previous_candle = dataframe.iloc[-2].squeeze()
+        if last_candle['close'] < previous_candle['close']:
+            return None
+
+        count_of_buys = 0
+        for order in trade.orders:
+            # Instantly stop when there's an open order
+            if order.ft_is_open:
+                return None
```

Comment:
I don't think we would be calling this callback if there is an open order.

Having this segment in the docs suggests that this is called every iteration, independent if there's open orders or not - which is not true.
The alternative would be to allow multiple open buy orders (some form of grid-orders i guess) - which i don't see handled anywhere - and i don't think is the purpose  / scope of this PR, and would be a little more complex, as then you'd have to remove `open_order_id` and instead check via the orders list - as there could be multiple orders at any moment.

Reply:
Fair point and we really shouldn't even call the method that calls the callback if there's a trade open. Code adjusted in the next commit.